plugins/modules/web_infrastructure/jenkins_build.py

```diff
@@ -216,21 +246,22 @@ def main():
             build_number=dict(type='int'),
             name=dict(required=True),
             password=dict(no_log=True),
-            state=dict(choices=['present', 'absent'], default="present"),
+            state=dict(choices=['present', 'absent', 'stopped'], default="present"),
             token=dict(no_log=True),
             url=dict(default="http://localhost:8080"),
             user=dict(),
         ),
-        mutually_exclusive=[
-            ['password', 'token'],
-        ],
+        mutually_exclusive=[['password', 'token']],
+        required_if=[['state', 'absent', ['build_number'], True], ['state', 'stopped', ['build_number'], True]],
```

Comment:
Checking thath `build_number` is present on `state=absent` should have a `bugfixes` entry in the changelog entry IMO.

Reply:
Before adding this check, without build number the module could handle this exception and return a fail json: 'jenkins.NotFoundException: Requested item could not be found'. So it is an improvement but probably not a bug fix?

Reply:
I would argue it is also a bugfix, since that message is very misleading: I would expect that message when I provided a build number that does not exist, or when the build has already been deleted (though I probably would not expect an error in that case, but rather `changed=false`), but not when I forgot to specify the build number.

Reply:
Sounds reasonable. I add it in the latest commit.